addons/l10n_co/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    document_type = fields.Selection([('rut', 'RUT'),
+                                      ('id_card', 'Tarjeta de Identidad'),
+                                      ('passport', 'Pasaporte'),
+                                      ('foreign_id_card', 'Cedula de Extranjeria'),
+                                      ('external_id', 'ID del Exterior')], string='Document Type')
+    verification_code = fields.Char(compute='_compute_verification_code', string='VC')
+
+    @api.depends('vat')
+    def _compute_verification_code(self):
+        for partner in self.filtered('vat'):
+            number = 0
+            multiplication_factors = [71, 67, 59, 53, 47, 43, 41, 37, 29, 23, 19, 17, 13, 7, 3]
+
+            for index, vat_number in enumerate(partner.vat[:len(multiplication_factors)]):
```

Comment:
Well, I didn't write this code but I guess it was to avoid potential IndexErrors when dealing with a `vat` that for some reason was longer than 15 chars. Instead I suppose it makes more sense to filter those out in the beginning because most likely the verification code wouldn't make sense anyway.